### PR TITLE
✨ aws.sqs.queues resource

### PR DESCRIPTION
**Pull request by vjeffrey**
Screenshot 2024-05-18 at 16 38 39

### CODE REVIEW

File: providers/aws/resources/aws.lr [hunk 1]
```diff
@@ -1836,6 +1836,46 @@ private aws.dynamodb.table @defaults("name region") {
   status string
 }
 
+// Amazon SQS
+aws.sqs {
+  // List of AWS SQS queues
```
Comment (tas50) [inline providers/aws/resources/aws.lr]:
```suggestion
  // List of Amazon SQS queues
```

File: providers/aws/resources/aws.lr [hunk 2]
```diff
@@ -1836,6 +1836,46 @@ private aws.dynamodb.table @defaults("name region") {
   status string
 }
 
+// Amazon SQS
```
Comment (tas50) [inline providers/aws/resources/aws.lr]:
```suggestion
// Amazon Simple Queue Service (SQS)
```

File: providers/aws/resources/aws.lr [hunk 3]
```diff
@@ -1836,6 +1836,46 @@ private aws.dynamodb.table @defaults("name region") {
...
+aws.sqs {
+  // List of AWS SQS queues
+  queues() []aws.sqs.queue
+}
+
+// Amazon SQS Queue
```
Comment (tas50) [inline providers/aws/resources/aws.lr]:
```suggestion
// Amazon Simple Queue Service (SQS) Queue
```

File: providers/aws/resources/aws.lr [hunk 4]
```diff
@@ -1836,6 +1836,46 @@ private aws.dynamodb.table @defaults("name region") {
...
+  maxReceiveCount() int
+  // Maximum message size for the queue
+  maximumMessageSize() int
+  // Time in seconds the queue retains messages
+  messageRetentionPeriodSeconds() int
+  // Time in seconds the queue waits for messages
+  receiveMessageWaitTimeSeconds() int
+  // Region for the queue
+  region string
+  // True if SSE is enabled for the queue
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // Whether SSE is enabled for the queue
```
(just our style)

File: providers/aws/resources/aws.lr [hunk 5]
```diff
@@ -1836,6 +1836,46 @@ private aws.dynamodb.table @defaults("name region") {
...
+  sqsManagedSseEnabled() bool
+  // Fifo or Standard
```
Comment (misterpantz) [inline providers/aws/resources/aws.lr]:
```suggestion
  // The type of queue: Fifo or Standard
```

**Review by misterpantz — COMMENTED** [review]
Docs look great! Just 2 suggestions :)

File: providers/aws/resources/aws_sqs.go [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+	err = json.Unmarshal([]byte(c), &r)
+	if err != nil {
+		return nil, err
+	}
+	parsedArn, err := arn.Parse(r.DeadLetterTargetArn)
+	if err != nil {
+		return nil, err
+	}
+	// "https://sqs.us-east-1.amazonaws.com/921877552404/Test-Preslav-Queue"
+	url := fmt.Sprintf("https://sqs.%s.amazonaws.com/%s/%s", a.Region.Data, "acctid", parsedArn.Resource)
```
Comment (preslavgerchev) [inline providers/aws/resources/aws_sqs.go]: there's a hardcoded "acctid" here, presume thats a testing leftover

Comment (vjeffrey) [reply providers/aws/resources/aws_sqs.go]: oh crap yes thank you!

**Review by preslavgerchev — APPROVED** [review]

**Pull request merged**